src/cmd/update.js

```diff
@@ -1,255 +1,44 @@
-		let openDelay = 0;
-		let openTimeout = OPEN_TIMEOUT;
-		let dev;
-		files = [...files];
-		try {
-			while (files.length) {
-				await delay(openDelay);
-				dev = await this._openDevice(deviceId, { timeout: openTimeout });
-				const file = files.shift();
-				if (file.useDfuMode) {
-					// Use DFU
-					if (!dev.isInDfuMode) {
-						// Disconnect the device from the network/cloud to ensure its system thread is unblocked
-						await this._disconnectDevice(dev);
-						await dev.enterDfuMode();
-					}
-					// Close the device before flashing it via dfu-util
-					await dev.close();
-					await dfu.writeModule(file.path, { vendorId: dev.vendorId, productId: dev.productId, serial: deviceId, leave: !files.length });
-					openDelay = 0;
-					openTimeout = OPEN_TIMEOUT;
-				} else {
-					// Use control requests
-					if (dev.isInDfuMode) {
-						await dev.reset();
-						await dev.close();
-						await delay(REOPEN_DELAY);
-						dev = await this._openDevice(deviceId);
-					}
-					await this._disconnectDevice(dev);
-					const data = fs.readFileSync(file.path);
-					await dev.updateFirmware(data, { timeout: FLASH_TIMEOUT });
-					await dev.close(); // Device is about to reset
-					openDelay = REOPEN_DELAY;
-					openTimeout = REOPEN_TIMEOUT;
-				}
-			}
-		} finally {
-			if (dev && dev.isOpen) {
-				try {
-					await dev.close();
-				} catch (err) {
-					// Ignore
-				}
-			}
-		}
+	constructor(...args) {
+		super(...args);
 	}
 
-	async _getDeviceInfo(idOrName) {
-		if (idOrName) {
-			const { api, auth } = this._particleApi();
-			const dev = await spin(openUsbDeviceByIdOrName(idOrName, api, auth, { dfuMode: true }), 'Getting device information...');
-			const devInfo = {
-				id: dev.id,
-				platformId: dev.platformId,
-				version: dev.firmwareVersion,
-				isInDfuMode: dev.isInDfuMode
-			};
-			await dev.close();
-			return devInfo;
-		}
-		const devs = await getUsbDevices({ dfuMode: true });
-		const devInfo = [];
-		for (const dev of devs) {
-			// Open the device to get its ID
-			await openUsbDevice(dev, { dfuMode: true });
-			devInfo.push({
-				id: dev.id,
-				platformId: dev.platformId,
-				version: dev.firmwareVersion,
-				isInDfuMode: dev.isInDfuMode
-			});
-			await dev.close();
-		}
-		if (!devInfo.length) {
-			throw new Error('No devices found');
-		}
-		if (devInfo.length === 1) {
-			return devInfo[0];
-		}
-		const answer = await prompt({
-			type: 'list',
-			name: 'device',
-			message: 'Which device would you like to update?',
-			choices: devInfo.map((d) => {
-				const platformName = isKnownPlatformId(d.platformId) ? platformForId(d.platformId).displayName : `Platform ${d.platformId}`;
-				return {
-					name: `${d.id} (${platformName})`,
-					value: d
-				};
-			})
+	async updateDevice(deviceIdOrName, { target }) {
+		const { api, auth , particleApi } = this._particleApi();
+		// get device info
+		const device = await usbUtils.getOneUsbDevice({ deviceIdOrName, api, auth, ui: this.ui });
```

Comment:
Same as this :)
https://github.com/particle-iot/particle-cli/pull/664

Reply:
Fixed here! https://github.com/particle-iot/particle-cli/pull/663/commits/49f313182606532d0000efb451c0c04265963262